openapi-v1.yaml

```diff
@@ -6156,3 +6278,124 @@ paths:
           description: error response
           schema:
             $ref: "#/definitions/Error"
+
+  /taskgraphs/log:
+    parameters:
+      - name: namespace
+        type: string
+        in: query
+        required: true
+        description: >
+          The namespace that owns the task graph.
+          This is required for POST requests to create a new graph.
+    post:
+      description: Create a task graph log.
+      operationId: createTaskGraphLog
+      tags:
+        - taskGraphLogs
+      responses:
+        201:
+          description: >
+            The task graph was created. The returned TaskGraphLog will include
+            the data the client sent, with the server-defined fields added in.
+          schema:
+            $ref: '#/definitions/TaskGraphLog'
+        default:
+          description: error response
+          schema:
+            $ref: '#/definitions/Error'
+
+  /taskgraphs/logs:
+    parameters:
+      - name: namespace
+        in: query
+        description: Include logs from only this namespace.
+        type: string
+        required: false
+      - name: created_by
+        in: query
+        description: Include logs from only this user.
+        type: string
+        required: false
+      - name: start_time
+        in: query
+        description: Include logs created after this time.
+        type: string
+        format: date-time
+        required: false
+      - name: end_time
+        in: query
+        description: Include logs created before this time.
+        type: string
+        format: date-time
+        required: false
+      - name: page
+        in: query
+        description: pagination offset
+        type: integer
+        required: false
+      - name: per_page
+        in: query
+        description: pagination limit
+        type: integer
+        required: false
+    get:
+      description: Fetch the task graph logs a user has access to.
+      operationId: listTaskGraphLogs
```

Comment:
Not sure about this. I understand it is provided for completeness of the API but i can't see how useful is for a user. It is needed by the UI for sure. Fine to keep it but i wouldn't object moving it to internal API

Reply:
I'm not sure that doing so would be worth the duplication and additional maintenance required, and I can see this being useful for the same reasons that task listing might be useful.